runtime/src/bank.rs

```diff
@@ -319,7 +319,7 @@ impl Bank {
             slots_per_year: parent.slots_per_year,
             epoch_schedule,
             rent_collector: parent.rent_collector.clone_with_epoch(epoch),
-            max_tick_height: (slot + 1) * parent.ticks_per_slot - 1,
+            max_tick_height: (slot + 1) * parent.ticks_per_slot,
```

Comment:
@mvines hehe this is basically what we had a few months ago before there was that huge rewrite to enforce that the first tick of every slot was equal to `0 mod ticks_per_slot` instead of `1 mod ticks_per_slot`. Do you recall why that was important?

Reply:
I forget.  This is just an off-by-one nightmare, whenever we have one the person who fixes it typically shifts from 0 -> 1 or 1->0 while claiming that was the problem all along.  This has happened multiple times :)

Reply:
Yeah the off-by-one stuff is a headache, are we open to shifting back (0->1) to remove the special handling of the first tick? Not sure if that's a compelling enough reason or not 😄

Reply:
I think it's possible to make it work with the shift to 0 -> 1, and it's nice that it simplifies the special case handling. I would run t through testnet with a lot of rigor before checking in though.

Reply:
I’m totally pro this direction too.  But yeah like Carl says, sorta expect to get backed out as I bet it’ll break something.   Nbd though. Now’s a good time to take a chance 🎰